--- a/setup.py	2015-10-19 19:18:26.000000000 +0300
+++ b/setup.py	2015-11-04 21:51:25.931115395 +0300
@@ -7,7 +7,7 @@
 
 requires = ['pyyaml>=3.11',
             'botocore>=1.0.1',
-            'cement==2.8.2',
+            'cement>=2.4',
-            'colorama==0.3.7',
+            'colorama>=0.3.3',
-            'pathspec==0.3.4',
+            'pathspec>=0.3.4',
             'setuptools >= 20.0',
@@ -17,27 +17,19 @@
-            'docker-py >= 1.1.0, <= 1.7.2',
+            'docker-py >= 1.1.0, < 1.9.0',
             'dockerpty >= 0.3.2, <= 0.4.1',
-            'semantic_version == 2.5.0'
+            'semantic_version >= 2.4.0'
            ]
 
 if not sys.platform.startswith('win'):
-    requires.append('blessed==1.9.5')
+    requires.append('blessed>=1.9.5')
 
-try:
-    with open('/etc/bash_completion.d/eb_completion.extra', 'w') as eo:
-        eo.write('')
-        data_files = [
-            ('/etc/bash_completion.d/', ['bin/eb_completion.bash'])
-        ]
-except:
-    # print('User does not have write access to /etc. Completion will not work.')
-    data_files = []
+data_files = []
 
 setup_options = dict(
     name='awsebcli',
     version=ebcli.__version__,
     description='Command Line Interface for AWS EB.',
     long_description=open('README.rst').read() + open('CHANGES.rst').read(),
-    scripts=['bin/eb_completion.bash'],
+    scripts=[],
     data_files=data_files,
     author='AWS Elastic Beanstalk',
     author_email='aws-eb-cli@amazon.com',
