From b7d46fe979f7b747ea27ad7b1059193674ff762e Mon Sep 17 00:00:00 2001
From: Azat Khuzhin <a3at.mail@gmail.com>
Date: Thu, 14 May 2020 20:42:32 +0300
Subject: [PATCH 2/3] Add missing headers for libstdc++ from gcc10

---
 src/Interpreters/addMissingDefaults.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Interpreters/addMissingDefaults.h b/src/Interpreters/addMissingDefaults.h
index d9d8d3d4f22..cc84f413b16 100644
--- a/dbms/src/Interpreters/addMissingDefaults.h
+++ b/dbms/src/Interpreters/addMissingDefaults.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <unordered_map>
+#include <string>
 
 
 namespace DB

From c059ee008519cff43b42022fc6f6a9d8ee86c56c Mon Sep 17 00:00:00 2001
From: Azat Khuzhin <a3at.mail@gmail.com>
Date: Thu, 14 May 2020 20:42:32 +0300
Subject: [PATCH 3/3] Use is_trivial+is_standard_layout over is_pod (later is
 deprecated)

---
 src/Common/Dwarf.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Common/Dwarf.cpp b/src/Common/Dwarf.cpp
index 6d3d8fdf0ee..7a697a2c9ef 100644
--- a/dbms/src/Common/Dwarf.cpp
+++ b/dbms/src/Common/Dwarf.cpp
@@ -104,7 +104,7 @@ namespace
 
 // Read (bitwise) one object of type T
 template <typename T>
-std::enable_if_t<std::is_pod_v<T>, T> read(std::string_view & sp)
+std::enable_if_t<std::is_trivial_v<T> && std::is_standard_layout_v<T>, T> read(std::string_view & sp)
 {
     SAFE_CHECK(sp.size() >= sizeof(T), "underflow");
     T x;

Fix (unsigned) cast
--- a/dbms/src/Common/Dwarf.cpp
+++ b/dbms/src/Common/Dwarf.cpp
@@ -447,14 +447,14 @@
         case DW_FORM_ref8:
             return read<uint64_t>(sp);
         case DW_FORM_sdata:
-            return readSLEB(sp);
+            return (uint64_t) readSLEB(sp);
         case DW_FORM_udata: [[fallthrough]];
         case DW_FORM_ref_udata:
             return readULEB(sp);
         case DW_FORM_flag:
             return read<uint8_t>(sp);
         case DW_FORM_flag_present:
-            return 1;
+            return 1u;
         case DW_FORM_sec_offset: [[fallthrough]];
         case DW_FORM_ref_addr:
             return readOffset(sp, is64Bit);
