diff --git a/cmake/Modules/FindgRPC.cmake b/cmake/Modules/FindgRPC.cmake
index 945d307952..d86e2addfd 100644
--- a/cmake/Modules/FindgRPC.cmake
+++ b/cmake/Modules/FindgRPC.cmake
@@ -248,6 +248,7 @@ find_library(gRPC_CPP_LIBRARY NAMES grpc++)
 find_library(gRPC_UNSECURE_LIBRARY NAMES grpc_unsecure)
 find_library(gRPC_CPP_UNSECURE_LIBRARY NAMES grpc++_unsecure)
 find_library(gRPC_CARES_LIBRARY NAMES cares)
+find_library(gRPC_GPR_LIBRARY NAMES gpr)
 
 set(gRPC_LIBRARIES)
 if(gRPC_USE_UNSECURE_LIBRARIES)
@@ -265,7 +266,7 @@ else()
     set(gRPC_LIBRARIES ${gRPC_LIBRARIES} ${gRPC_CPP_LIBRARY})
   endif()
 endif()
-set(gRPC_LIBRARIES ${gRPC_LIBRARIES} ${gRPC_CARES_LIBRARY})
+set(gRPC_LIBRARIES ${gRPC_LIBRARIES} ${gRPC_GPR_LIBRARY} ${gRPC_CARES_LIBRARY})
 
 # Restore the original find library ordering.
 if(gRPC_USE_STATIC_LIBS)
