diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b80bcfdf4d..4686cf4cee 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -361,10 +361,8 @@ if (ZSTD_LIBRARY)
 endif()
 
-set (LZMA_LIBRARY liblzma)
-set (LZMA_INCLUDE_DIR ${ClickHouse_SOURCE_DIR}/contrib/xz/src/liblzma/api)
+set (LZMA_LIBRARY lzma)
 if (LZMA_LIBRARY)
     target_link_libraries (clickhouse_common_io PUBLIC ${LZMA_LIBRARY})
-    target_include_directories (clickhouse_common_io SYSTEM BEFORE PUBLIC ${LZMA_INCLUDE_DIR})
 endif()
 
 if (USE_ICU)
--- a/contrib/CMakeLists.txt
+++ b/contrib/CMakeLists.txt
@@ -47,7 +47,6 @@
 add_subdirectory (murmurhash)
 add_subdirectory (replxx-cmake)
 add_subdirectory (unixodbc-cmake)
-add_subdirectory (xz)
 
 add_subdirectory (poco-cmake)
 add_subdirectory (croaring-cmake)
