diff --git a/base/ext/bit_cast.h b/base/ext/bit_cast.h
index d96ddb7590..68ad99a0f2 100644
--- a/base/ext/bit_cast.h
+++ b/base/ext/bit_cast.h
@@ -25,6 +25,6 @@ namespace ext
     std::decay_t<To> safe_bit_cast(const From & from)
     {
         static_assert(sizeof(To) == sizeof(From), "bit cast on types of different width");
-        return bit_cast<To, From>(from);
+        return ext::bit_cast<To, From>(from);
     }
 }
diff --git a/src/Columns/Collator.h b/src/Columns/Collator.h
index df60aaba43..1892ccdd5e 100644
--- a/src/Columns/Collator.h
+++ b/src/Columns/Collator.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <optional>
 #include <string>
 #include <vector>
 #include <boost/noncopyable.hpp>
diff --git a/src/Core/NamesAndTypes.h b/src/Core/NamesAndTypes.h
index dad031a543..1688be3c3e 100644
--- a/src/Core/NamesAndTypes.h
+++ b/src/Core/NamesAndTypes.h
@@ -2,6 +2,7 @@
 
 #include <map>
 #include <list>
+#include <optional>
 #include <string>
 #include <set>
 #include <initializer_list>
diff --git a/src/IO/HashingWriteBuffer.h b/src/IO/HashingWriteBuffer.h
index 539c0390dc..e3132fcef8 100644
--- a/src/IO/HashingWriteBuffer.h
+++ b/src/IO/HashingWriteBuffer.h
@@ -17,11 +17,16 @@ class IHashingBuffer : public BufferWithOwnMemory<Buffer>
 public:
     using uint128 = CityHash_v1_0_2::uint128;
 
-    IHashingBuffer<Buffer>(size_t block_size_ = DBMS_DEFAULT_HASHING_BLOCK_SIZE)
+    IHashingBuffer(size_t block_size_)
         : BufferWithOwnMemory<Buffer>(block_size_), block_pos(0), block_size(block_size_), state(0, 0)
     {
     }
 
+    IHashingBuffer()
+        : IHashingBuffer<Buffer>(DBMS_DEFAULT_HASHING_BLOCK_SIZE)
+    {
+    }
+
     uint128 getHash()
     {
         if (block_pos)
diff --git a/src/Interpreters/DatabaseCatalog.h b/src/Interpreters/DatabaseCatalog.h
index bb82dbfc44..81a1f94925 100644
--- a/src/Interpreters/DatabaseCatalog.h
+++ b/src/Interpreters/DatabaseCatalog.h
@@ -13,6 +13,7 @@
 #include <mutex>
 #include <shared_mutex>
 #include <array>
+#include <condition_variable>
 #include <list>
 
 
diff --git a/src/Server/grpc_protos/CMakeLists.txt b/src/Server/grpc_protos/CMakeLists.txt
index 3661d6083b..be135792d7 100644
--- a/src/Server/grpc_protos/CMakeLists.txt
+++ b/src/Server/grpc_protos/CMakeLists.txt
@@ -8,4 +8,4 @@ set (CMAKE_CXX_CLANG_TIDY "")
 
 add_library(clickhouse_grpc_protos STATIC ${clickhouse_grpc_proto_headers} ${clickhouse_grpc_proto_sources})
 target_include_directories(clickhouse_grpc_protos SYSTEM PUBLIC ${gRPC_INCLUDE_DIRS} ${Protobuf_INCLUDE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
-target_link_libraries (clickhouse_grpc_protos PUBLIC ${gRPC_LIBRARIES})
+target_link_libraries (clickhouse_grpc_protos PUBLIC ${gRPC_LIBRARIES} ${Protobuf_LIBRARIES})
diff --git a/src/Storages/MergeTree/MergeTreePartInfo.h b/src/Storages/MergeTree/MergeTreePartInfo.h
index 66d5342b67..e928ee92cf 100644
--- a/src/Storages/MergeTree/MergeTreePartInfo.h
+++ b/src/Storages/MergeTree/MergeTreePartInfo.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <limits>
 #include <tuple>
 #include <common/types.h>
 #include <common/DayNum.h>
