diff --git a/programs/CMakeLists.txt b/programs/CMakeLists.txt
index 89220251cd..bd2d23a3fb 100644
--- a/programs/CMakeLists.txt
+++ b/programs/CMakeLists.txt
@@ -18,6 +18,8 @@ option (ENABLE_CLICKHOUSE_FORMAT "Enable clickhouse-format" ${ENABLE_CLICKHOUSE_
 option (ENABLE_CLICKHOUSE_OBFUSCATOR "Enable clickhouse-obfuscator" ${ENABLE_CLICKHOUSE_ALL})
 option (ENABLE_CLICKHOUSE_ODBC_BRIDGE "Enable clickhouse-odbc-bridge" ${ENABLE_CLICKHOUSE_ALL})
 
+include (../cmake/version.cmake)
+
 if (CLICKHOUSE_SPLIT_BINARY)
     option (ENABLE_CLICKHOUSE_INSTALL "Enable clickhouse-install" OFF)
 else ()
