diff --git a/cmake/find/s3.cmake b/cmake/find/s3.cmake
index 1bbf48fd6b..72f290a626 100644
--- a/cmake/find/s3.cmake
+++ b/cmake/find/s3.cmake
@@ -19,14 +19,12 @@ if (NOT USE_INTERNAL_POCO_LIBRARY AND USE_INTERNAL_AWS_S3_LIBRARY)
 endif()
 
 if (NOT USE_INTERNAL_AWS_S3_LIBRARY)
-    message (${RECONFIGURE_MESSAGE_LEVEL} "Compilation with external S3 library is not supported yet")
+    find_package(AWSSDK REQUIRED COMPONENTS core s3)
 endif()
 
-if (NOT EXISTS "${ClickHouse_SOURCE_DIR}/contrib/aws/aws-cpp-sdk-s3")
+if (USE_INTERNAL_AWS_S3_LIBRARY AND NOT EXISTS "${ClickHouse_SOURCE_DIR}/contrib/aws/aws-cpp-sdk-s3")
     message (WARNING "submodule contrib/aws is missing. to fix try run: \n git submodule update --init --recursive")
-    if (USE_INTERNAL_AWS_S3_LIBRARY)
-        message (${RECONFIGURE_MESSAGE_LEVEL} "Can't find internal S3 library")
-    endif ()
+    message (${RECONFIGURE_MESSAGE_LEVEL} "Can't find internal S3 library")
     set (MISSING_AWS_S3 1)
 endif ()

@@ -36,7 +36,7 @@ if (USE_INTERNAL_AWS_S3_LIBRARY AND NOT MISSING_AWS_S3)
     set(AWS_S3_LIBRARY aws_s3)
     set(USE_INTERNAL_AWS_S3_LIBRARY 1)
     set(USE_AWS_S3 1)
-else()
+elseif(MISSING_AWS_S3)
     message (${RECONFIGURE_MESSAGE_LEVEL} "Can't enable S3")
     set(USE_INTERNAL_AWS_S3_LIBRARY 0)
     set(USE_AWS_S3 0)
 
