diff --git a/contrib/poco-cmake/Crypto/CMakeLists.txt b/contrib/poco-cmake/Crypto/CMakeLists.txt
index 468f164151..1685e96728 100644
--- a/contrib/poco-cmake/Crypto/CMakeLists.txt
+++ b/contrib/poco-cmake/Crypto/CMakeLists.txt
@@ -30,7 +30,7 @@ if (ENABLE_SSL)
 
         target_compile_options (_poco_crypto PRIVATE -Wno-newline-eof)
         target_include_directories (_poco_crypto SYSTEM PUBLIC ${LIBRARY_DIR}/Crypto/include)
-        target_link_libraries (_poco_crypto PUBLIC Poco::Foundation ssl)
+        target_link_libraries (_poco_crypto PUBLIC Poco::Foundation ssl crypto)
     else ()
         add_library (Poco::Crypto UNKNOWN IMPORTED GLOBAL)
 
