commit 7e2b3b59b8bccbe1336673ab8a2a65a48f260aad
Author: Danila Kutenin <danlark@sandbox-286092131>
Date:   Fri May 31 11:55:01 2019 +0300

    Fix gcc9 warnings and ICE's

diff --git a/dbms/src/Storages/ColumnDefault.cpp b/dbms/src/Storages/ColumnDefault.cpp
index 2610b14f0..d35d3c314 100644
--- a/dbms/src/Storages/ColumnDefault.cpp
+++ b/dbms/src/Storages/ColumnDefault.cpp
@@ -31,7 +31,10 @@ ColumnDefaultKind columnDefaultKindFromString(const std::string & str)
     };
 
     const auto it = map.find(str);
-    return it != std::end(map) ? it->second : throw Exception{"Unknown column default specifier: " + str, ErrorCodes::LOGICAL_ERROR};
+    if (it != std::end(map))
+        return it->second;
+
+    throw Exception{"Unknown column default specifier: " + str, ErrorCodes::LOGICAL_ERROR};
 }
 
 
@@ -44,7 +47,10 @@ std::string toString(const ColumnDefaultKind kind)
     };
 
     const auto it = map.find(kind);
-    return it != std::end(map) ? it->second : throw Exception{"Invalid ColumnDefaultKind", ErrorCodes::LOGICAL_ERROR};
+    if (it != std::end(map))
+        return it->second;
+
+    throw Exception{"Invalid ColumnDefaultKind", ErrorCodes::LOGICAL_ERROR};
 }

