diff --git a/programs/benchmark/CMakeLists.txt b/programs/benchmark/CMakeLists.txt
index 3fa8deb6bd..3afa14c240 100644
--- a/programs/benchmark/CMakeLists.txt
+++ b/programs/benchmark/CMakeLists.txt
@@ -1,4 +1,4 @@
-set (CLICKHOUSE_BENCHMARK_SOURCES Benchmark.cpp)
+set (CLICKHOUSE_BENCHMARK_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/Benchmark.cpp)
 
 set (CLICKHOUSE_BENCHMARK_LINK
     PRIVATE
diff --git a/programs/client/CMakeLists.txt b/programs/client/CMakeLists.txt
index 72b5caf978..d89c49e0c8 100644
--- a/programs/client/CMakeLists.txt
+++ b/programs/client/CMakeLists.txt
@@ -1,8 +1,8 @@
 set (CLICKHOUSE_CLIENT_SOURCES
-    Client.cpp
-    ConnectionParameters.cpp
-    QueryFuzzer.cpp
-    Suggest.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/Client.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/ConnectionParameters.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/QueryFuzzer.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/Suggest.cpp
 )
 
 set (CLICKHOUSE_CLIENT_LINK
diff --git a/programs/compressor/CMakeLists.txt b/programs/compressor/CMakeLists.txt
index ff642a32fd..4eac7aa513 100644
--- a/programs/compressor/CMakeLists.txt
+++ b/programs/compressor/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Also in utils
 
-set (CLICKHOUSE_COMPRESSOR_SOURCES Compressor.cpp)
+set (CLICKHOUSE_COMPRESSOR_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/Compressor.cpp)
 
 set (CLICKHOUSE_COMPRESSOR_LINK
     PRIVATE
diff --git a/programs/extract-from-config/CMakeLists.txt b/programs/extract-from-config/CMakeLists.txt
index ff2d793711..a7abd92bae 100644
--- a/programs/extract-from-config/CMakeLists.txt
+++ b/programs/extract-from-config/CMakeLists.txt
@@ -1,4 +1,4 @@
-set (CLICKHOUSE_EXTRACT_FROM_CONFIG_SOURCES ExtractFromConfig.cpp)
+set (CLICKHOUSE_EXTRACT_FROM_CONFIG_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/ExtractFromConfig.cpp)
 
 set (CLICKHOUSE_EXTRACT_FROM_CONFIG_LINK
     PRIVATE
diff --git a/programs/format/CMakeLists.txt b/programs/format/CMakeLists.txt
index 49f17ef163..b5db7fb143 100644
--- a/programs/format/CMakeLists.txt
+++ b/programs/format/CMakeLists.txt
@@ -1,4 +1,4 @@
-set (CLICKHOUSE_FORMAT_SOURCES Format.cpp)
+set (CLICKHOUSE_FORMAT_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/Format.cpp)
 
 set (CLICKHOUSE_FORMAT_LINK
     PRIVATE
diff --git a/programs/git-import/CMakeLists.txt b/programs/git-import/CMakeLists.txt
index 279bb35a27..1eb37b1ca9 100644
--- a/programs/git-import/CMakeLists.txt
+++ b/programs/git-import/CMakeLists.txt
@@ -1,4 +1,4 @@
-set (CLICKHOUSE_GIT_IMPORT_SOURCES git-import.cpp)
+set (CLICKHOUSE_GIT_IMPORT_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/git-import.cpp)
 
 set (CLICKHOUSE_GIT_IMPORT_LINK
     PRIVATE
diff --git a/programs/install/CMakeLists.txt b/programs/install/CMakeLists.txt
index c3f4d96d63..021f7dbe97 100644
--- a/programs/install/CMakeLists.txt
+++ b/programs/install/CMakeLists.txt
@@ -1,4 +1,4 @@
-set (CLICKHOUSE_INSTALL_SOURCES Install.cpp)
+set (CLICKHOUSE_INSTALL_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/Install.cpp)
 
 set (CLICKHOUSE_INSTALL_LINK
     PRIVATE
diff --git a/programs/local/CMakeLists.txt b/programs/local/CMakeLists.txt
index b61f0ea33b..01c9772e34 100644
--- a/programs/local/CMakeLists.txt
+++ b/programs/local/CMakeLists.txt
@@ -1,4 +1,4 @@
-set (CLICKHOUSE_LOCAL_SOURCES LocalServer.cpp)
+set (CLICKHOUSE_LOCAL_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/LocalServer.cpp)
 
 set (CLICKHOUSE_LOCAL_LINK
     PRIVATE
diff --git a/programs/obfuscator/CMakeLists.txt b/programs/obfuscator/CMakeLists.txt
index d1179b3718..c67b3c00dd 100644
--- a/programs/obfuscator/CMakeLists.txt
+++ b/programs/obfuscator/CMakeLists.txt
@@ -1,4 +1,4 @@
-set (CLICKHOUSE_OBFUSCATOR_SOURCES Obfuscator.cpp)
+set (CLICKHOUSE_OBFUSCATOR_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/Obfuscator.cpp)
 
 set (CLICKHOUSE_OBFUSCATOR_LINK
     PRIVATE
diff --git a/programs/odbc-bridge/CMakeLists.txt b/programs/odbc-bridge/CMakeLists.txt
index 1186435461..557d55105f 100644
--- a/programs/odbc-bridge/CMakeLists.txt
+++ b/programs/odbc-bridge/CMakeLists.txt
@@ -1,16 +1,16 @@
 set (CLICKHOUSE_ODBC_BRIDGE_SOURCES
-    ColumnInfoHandler.cpp
-    getIdentifierQuote.cpp
-    HandlerFactory.cpp
-    IdentifierQuoteHandler.cpp
-    MainHandler.cpp
-    ODBCBlockInputStream.cpp
-    ODBCBlockOutputStream.cpp
-    ODBCBridge.cpp
-    PingHandler.cpp
-    SchemaAllowedHandler.cpp
-    validateODBCConnectionString.cpp
-    odbc-bridge.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/ColumnInfoHandler.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/getIdentifierQuote.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/HandlerFactory.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/IdentifierQuoteHandler.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/MainHandler.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/ODBCBlockInputStream.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/ODBCBlockOutputStream.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/ODBCBridge.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/PingHandler.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/SchemaAllowedHandler.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/validateODBCConnectionString.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/odbc-bridge.cpp
 )
 
 if (OS_LINUX)
diff --git a/programs/server/CMakeLists.txt b/programs/server/CMakeLists.txt
index 198d908116..c9bcd8b0d3 100644
--- a/programs/server/CMakeLists.txt
+++ b/programs/server/CMakeLists.txt
@@ -1,6 +1,6 @@
 set(CLICKHOUSE_SERVER_SOURCES
-    MetricsTransmitter.cpp
-    Server.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/MetricsTransmitter.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/Server.cpp
 )
 
 if (OS_LINUX)
@@ -52,10 +52,4 @@ if (OS_LINUX)
 
     add_library(clickhouse_server_configs STATIC ${RESOURCE_OBJS})
     set_target_properties(clickhouse_server_configs PROPERTIES LINKER_LANGUAGE C)
-
-    # whole-archive prevents symbols from being discarded for unknown reason
-    # CMake can shuffle each of target_link_libraries arguments with other
-    # libraries in linker command. To avoid this we hardcode whole-archive
-    # library into single string.
-    add_dependencies(clickhouse-server-lib clickhouse_server_configs)
 endif ()
