--- a/third_party/llvm-project/include/llvm/ADT/edit_distance.h
+++ b/third_party/llvm-project/include/llvm/ADT/edit_distance.h
@@ -58,7 +58,7 @@
   typename ArrayRef<T>::size_type n = ToArray.size();
 
   const unsigned SmallBufferSize = 64;
-  unsigned SmallBuffer[SmallBufferSize];
+  unsigned SmallBuffer[SmallBufferSize] = {0};
   std::unique_ptr<unsigned[]> Allocated;
   unsigned *Row = SmallBuffer;
   if (n + 1 > SmallBufferSize) {
