From 45829bea9a1eb9ef2f937823f9c9e6bd583400e6 Mon Sep 17 00:00:00 2001
From: takatakizashi <20noir@gmail.com>
Date: Sun, 25 Aug 2019 00:33:00 +0900
Subject: [PATCH] Update urllib3's version of install_requires to match
 requests

---
 setup.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index f5c4dd9..45d4f38 100644
--- a/setup.py
+++ b/setup.py
@@ -31,9 +31,8 @@
     keywords=['Kaggle', 'API'],
     entry_points={'console_scripts': ['kaggle = kaggle.cli:main']},
     install_requires=[
-        # Restriction that urllib3's version is less than 1.25 needed to avoid
-        # requests dependency problem.
-        'urllib3 >= 1.21.1, < 1.25',
+        # Restriction that urllib3's version needed to avoid requests dependency problem.
+        'urllib3 >= 1.21.1, < 1.26, != 1.25.0, != 1.25.1',
         'six >= 1.10',
         'certifi',
         'python-dateutil',
