--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -588,7 +588,7 @@
 
     cflags += [ "-flto=thin" ]
 
-    if (target_os != "chromeos") {
+    if (target_os != "chromeos" && is_clang) {
       cflags += [ "-fsplit-lto-unit" ]
     }
 
@@ -655,7 +655,7 @@
 
     # TODO(pcc): Re-enable this flag on Android. This will require libc++ to be
     # built with ThinLTO (see https://crbug.com/767901) as well as the GVR shim.
-    if (!is_android) {
+    if (!is_android && is_clang) {
       cflags += [ "-fwhole-program-vtables" ]
       if (!is_win) {
         ldflags += [ "-fwhole-program-vtables" ]
