--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -270,7 +270,6 @@ config("compiler") {
 
   configs += [
     # See the definitions below.
-    ":clang_revision",
     ":compiler_cpu_abi",
     ":compiler_codegen",
     ":compiler_deterministic",
@@ -742,7 +741,6 @@ config("compiler_cpu_abi") {
     if (current_cpu == "x64") {
       cflags += [
         "-m64",
-        "-march=$x64_arch",
       ]
       ldflags += [ "-m64" ]
     } else if (current_cpu == "x86") {
@@ -2002,7 +2000,7 @@ config("optimize") {
       cflags = [ "-Os" ] + common_optimize_on_cflags
     }
   } else {
-    cflags = [ "-O2" ] + common_optimize_on_cflags
+    cflags = common_optimize_on_cflags
   }
   ldflags = common_optimize_on_ldflags
 }
@@ -2062,7 +2060,7 @@ config("optimize_max") {
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-O2" ] + common_optimize_on_cflags
+      cflags = common_optimize_on_cflags
     }
   }
 }
@@ -2094,7 +2092,7 @@ config("optimize_speed") {
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-O3" ] + common_optimize_on_cflags
+      cflags = common_optimize_on_cflags
     }
   }
 }
